datastore/simple/logger_test.go

```diff
@@ -14,51 +15,28 @@ import (
 var _ = Describe("Proto Logger", func() {
 	var (
 		protoLogger *simple.ProtoLogger
+		logger      *logrus.Logger
+		hook        *test.Hook
 	)
 
 	BeforeEach(func() {
-		logger, _ := test.NewNullLogger()
+		logger, hook = test.NewNullLogger()
 		protoLogger, _ = simple.NewProtoLogger(logger).(*simple.ProtoLogger)
 	})
 
-	DescribeTable("Proto Parsing",
-		func(txType string, input proto.Message, verifyOutput func(proto.Message) bool) {
+	DescribeTable("Produce",
+		func(txType string, input proto.Message, logContains string) {
 			payloadBytes, err := proto.Marshal(input)
 			Expect(err).NotTo(HaveOccurred())
-			output, err := protoLogger.GetProtoMessage(&telemetry.Record{
+			protoLogger.Produce(&telemetry.Record{
 				TxType:       txType,
 				PayloadBytes: payloadBytes,
 			})
-			Expect(err).NotTo(HaveOccurred())
-			Expect(verifyOutput(output)).To(BeTrue())
+			Expect(hook.LastEntry().Message).To(ContainSubstring(logContains))
 		},
-		Entry("for txType alerts", "alerts", &protos.VehicleAlerts{Vin: "testAlertVin"}, func(msg proto.Message) bool {
-			myMsg, ok := msg.(*protos.VehicleAlerts)
-			if !ok {
-				return false
-			}
-			return myMsg.GetVin() == "testAlertVin"
-		}),
-		Entry("for txType errors", "errors", &protos.VehicleErrors{Vin: "testErrorVin"}, func(msg proto.Message) bool {
-			myMsg, ok := msg.(*protos.VehicleErrors)
-			if !ok {
-				return false
-			}
-			return myMsg.GetVin() == "testErrorVin"
-		}),
-		Entry("for txType V", "V", &protos.Payload{Vin: "testPayloadVIN"}, func(msg proto.Message) bool {
-			myMsg, ok := msg.(*protos.Payload)
-			if !ok {
-				return false
-			}
-			return myMsg.GetVin() == "testPayloadVIN"
-		}),
+		Entry("for txType alerts", "alerts", &protos.VehicleAlerts{Vin: "testAlertVin"}, "logger_json_unmarshal"),
```

Comment:
we should still validate for `testAlertVin` is present in the hook and not `logger_json_unmarshal`?

Reply:
Updated to validate the expected json is included in log.